s3.tf

```diff
@@ -0,0 +1,44 @@
+
+resource "aws_s3_bucket" "codebuild_artifacts" {
+  bucket = local.project_name_s3  # S3 bucket for CodeBuild artifacts
+
+  force_destroy = true  # Force delete the bucket even if it has objects
+
+  tags = var.tags  # Apply tags
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "codebuild_artifacts_encryption" {
+  bucket = aws_s3_bucket.codebuild_artifacts.bucket
+
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm     = "aws:kms"  # Use AWS KMS for encryption
+      kms_master_key_id = "alias/aws/s3"  # Default KMS key for S3
+    }
+  }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "codebuild_artifacts_lifecycle" {
+  bucket = aws_s3_bucket.codebuild_artifacts.bucket
+
+  rule {
+    id     = "lifecycle-rule"  # Lifecycle rule ID
+    status = "Enabled"  # Enable the lifecycle rule
+
+    expiration {
+      days = 90  # Expire objects after 90 days
```

Comment:
I don't think that's a good idea - what if I'd like to rerun the pipeline after 3 months?